conn.go

```diff
@@ -1046,14 +1045,14 @@ func (c *conn) resetTransactionForRetry(ctx context.Context, errDuringCommit boo
 	return c.tx.resetForRetry(ctx)
 }
 
+func (c *conn) withTransactionCloseFunc(close func()) {
+	c.tempTransactionCloseFunc = close
+}
+
 func (c *conn) withTempTransactionOptions(options *ReadWriteTransactionOptions) {
 	if options == nil {
 		return
 	}
-	c.tempTransactionCloseFunc = options.close
-	// Start a transaction for the connection state, so we can set the transaction options
-	// as local options in the current transaction.
-	_ = c.state.Begin()
```

Comment:
This has been moved into the `BeginTx` function. The flow is now:
1. BeginTx -> This starts the transaction for the connection state and creates an 'empty' transaction that can become both a read-only or a read/write transaction at a later moment. The `close` function is an argument for the `BeginTx` function.
2. withTempTransactionOptions is called __after__ the call to `BeginTx`. The transaction options are then stored in the local session state of that transaction.
3. The application can set further transaction options by executing for example `set transaction read only`
4. When the first query or DML statement is executed on the transaction, the actual transaction is started. From this point on, it is no longer possible to set further transaction options.